palace/cmake/PkgConfigHelpers.cmake

```diff
@@ -127,17 +128,18 @@ int main()
     PETSC_TEST_COMPILED
     ${PETSC_LIB_TEST_DIR}
     ${PETSC_LIB_TEST_CPP}
+    CMAKE_FLAGS -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
```

Comment:
@hughcars you left a comment above about the Spack CMake flag addition. I can double check, but I needed both the spack modification and the modification to the CMake here in order to get Spack builds working and using the intended compiler toolchain.